CHANGELOG 2.0.0-beta.17 missing changes (#4579)
mapbox/mapbox-navigation-android: CHANGELOG.md

- Added new `HistoryEvent`: `HistoryEventPushHistoryRecord` allows read events set with **type**-**properties** structure. [#4567](https://github.com/mapbox/mapbox-navigation-android/pull/4567)
- :warning: Added new APIs to request and render road shields asynchronously, see `MapboxManeuverApi#getRoadShields` and `MapboxManeuverView#renderManeuverShields` for details. Shields are not automatically downloaded anymore to allow for more granular and faster initial updates. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)
- Exposed `RouteStepProgress#instructionIndex` that indicates which instruction out of the list of instructions for a step is the currently active one. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)

> - Refactored PredictiveCacheController to handle multiple map instances, add ability to filter sources to cache. [#4539](https://github.com/mapbox/mapbox-navigation-android/pull/4539)

Reviewer: ```suggestion
- :warning: Refactored `PredictiveCacheController` to handle multiple map instances, add the ability to filter sources to cache. [#4539](https://github.com/mapbox/mapbox-navigation-android/pull/4539)
```
and merge the below line items into this entry.

#### Bug fixes and improvements
- Fixed a bug where in some cases status updates were not generated after switching between offline and online states. [#4558](https://github.com/mapbox/mapbox-navigation-android/pull/4558)
- Fixed an issue where `RouteOptions` weren't appended to onboard router results which could've caused incorrect reroutes that missed silent waypoints or other waypoint modifiers. [#4563](https://github.com/mapbox/mapbox-navigation-android/pull/4563)
- Refactored dependency of `libnavui-resources` by changing from api to implementation. [#4568](https://github.com/mapbox/mapbox-navigation-android/pull/4568)
- :warning: `RouteProgress#bannerInstructions` does not become `null` and keeps its reference for as long as a step is active. Listen to `BannerInstructionsObserver` if you need a specific timing event when the banner first appears. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)

> - :warning: `fun setMapInstance(map: MapboxMap)` is replaced with `fun addMapInstance(map: MapboxMap,  sourceIdsToCache: List<String>)`.  [#4539](https://github.com/mapbox/mapbox-navigation-android/pull/4539)

Reviewer: ```suggestion
- :warning: `fun setMapInstance(map: MapboxMap)` is replaced with `fun addMapInstance(map: MapboxMap,  sourceIdsToCache: List<String>)` to allow for more granular caching configuration and support multiple map instances at the same time.  [#4539](https://github.com/mapbox/mapbox-navigation-android/pull/4539)
```

Reviewer: This and all below should ideally be in the same line entry as the "Refactored PredictiveCacheController to handle multiple map instances..." feature above. It all impacts the same piece of code, I don't feel like we need to make multiple entries.

- :warning: `fun removeMapInstance()` is replaced with `fun removeMapInstance(map: MapboxMap)`.  [#4539](https://github.com/mapbox/mapbox-navigation-android/pull/4539)

> - :warning: `predictiveCacheLocationOptions: PredictiveCacheLocationOptions` is removed from `NavigationOptions`.  [#4539](https://github.com/mapbox/mapbox-navigation-android/pull/4539)

Reviewer: ```suggestion
- :warning: `predictiveCacheLocationOptions: PredictiveCacheLocationOptions` is moved from `NavigationOptions` to `PredictiveCacheController. [#4539](https://github.com/mapbox/mapbox-navigation-android/pull/4539)
```

> - :warning: map instance has to be configured with the same [TileStore] instance that was provided to [RoutingTilesOptions.tileStore].  [#4539](https://github.com/mapbox/mapbox-navigation-android/pull/4539)

Reviewer: ```suggestion
- :warning: Note: The map instance has to be configured with the same [TileStore] instance that was provided to [RoutingTilesOptions.tileStore] to best support predictive caching and offline features.  [#4539](https://github.com/mapbox/mapbox-navigation-android/pull/4539)
```

- :warning: Added new APIs to request and render road shields asynchronously, see `MapboxManeuverApi#getRoadShields` and `MapboxManeuverView#renderManeuverShields` for details. Shields are not automatically downloaded anymore to allow for more granular and faster initial updates. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)
- Exposed `RouteStepProgress#instructionIndex` that indicates which instruction out of the list of instructions for a step is the currently active one. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)
- :warning: Refactored `PredictiveCacheController` to handle multiple map instances, add the ability to filter sources to cache. [#4539](https://github.com/mapbox/mapbox-navigation-android/pull/4539)
    - `fun setMapInstance(map: MapboxMap)` is replaced with `fun addMapInstance(map: MapboxMap,  sourceIdsToCache: List<String>)` to allow for more granular caching configuration and support multiple map instances at the same time.
    - `fun removeMapInstance()` is replaced with `fun removeMapInstance(map: MapboxMap)`.

>     - `predictiveCacheLocationOptions: PredictiveCacheLocationOptions` is moved from `NavigationOptions` to `PredictiveCacheController. 

Reviewer: ```suggestion
    - `predictiveCacheLocationOptions: PredictiveCacheLocationOptions` is moved from `NavigationOptions` to `PredictiveCacheController`. 
```

>     -  The map instance has to be configured with the same [TileStore] instance that was provided to [RoutingTilesOptions.tileStore] to best support predictive caching and offline features.

Reviewer: ```suggestion
    -  Note: the map instance has to be configured with the same `TileStore` instance that was provided to `RoutingTilesOptions.tileStore` to best support predictive caching and offline features.
```
